pkg/3scale/amp/operator/system_searchd_options_provider.go

```diff
@@ -80,6 +80,13 @@ func (s *SystemSearchdOptionsProvider) podTemplateLabels() map[string]string {
 		labels[k] = v
 	}
 
+	if s.apimanager.Spec.System.SearchdSpec != nil &&
```

Comment:
this `if` is not needed. `s.apimanager.Spec.System.SearchdSpec` will never be `nil` at this context and no need to check about `s.apimanager.Spec.System.SearchdSpec.Labels`